definitions/procedures/maintenance_mode/enable_maintenance_mode.rb

```diff
@@ -0,0 +1,39 @@
+module Procedures::MaintenanceMode
+  class EnableMaintenanceMode < ForemanMaintain::Procedure
+    metadata do
+      label :enable_maintenance_mode
+      description 'Add maintenance_mode tables/chain to nftables/iptables'
+      tags :pre_migrations, :maintenance_mode_on
+      after :sync_plans_disable
+      confine do
+        feature(:nftables) || feature(:iptables)
+      end
+    end
+
+    def run
+      if feature(:nftables)
+        nftables_enable_maintenance_mode
+      elsif feature(:iptables)
+        feature(:iptables).add_maintenance_mode_chain
+      else
+        warn! 'Unable to find iptables or nftables!'
+      end
```

Comment:
same comment as above, I think both features should implement the same interface, and we should call that interface from the procedure

Reply:
much more readable,thanks!